tests/integration/components/make-decision-dropdown/component-collection-test.ts

```diff
@@ -0,0 +1,168 @@
+            checkboxLabel: ['Remove submission'],
+            // eslint-disable-next-line max-len
+            checkboxDescription: ['Collection will be removed from the collection and no longer appear in search results'],
+        },
+        [CollectionSubmissionReviewStates.InProgress]: {
+            actions: [],
+        },
+        [CollectionSubmissionReviewStates.Rejected]: {
+            actions: [],
+        },
+        [CollectionSubmissionReviewStates.Removed]: {
+            actions: [],
+        },
+    };
+
+    async function buildCollectionScenarios(
+        testContext: TestContext,
+        reviewsState: CollectionSubmissionReviewStates,
+    ): Promise<void> {
+        // Given the scenario needs to be created for a collection
+        // to have a submission request in a certain review state
+
+        // When the fixture loaded
+        server.loadFixtures('licenses');
+        // And a user is created
+        const currentUser = server.create('user', 'loggedIn');
+        // And licenses are addee
+        const licensesAcceptable = server.schema.licenses.all().models;
+        // And a collection provider is created
+        const provider = server.create('collection-provider', 'currentUserIsModerator');
+        // And a collection is added with the provider
+        const mirageCollection = server.create('collection', {
+            provider,
+        });
+        // And a node with a guid is created
+        const guid = server.create('node', {
+            license: licensesAcceptable[0],
+            currentUserPermissions: [Permission.Admin],
+        }, 'withContributors');
+
+        // And a contributor is added
+        server.create('contributor', {
+            node: guid,
+            users: currentUser,
+            index: 0,
+        });
+        // And a moderator is assignede
+        server.create('moderator', { provider });
+        // And a collection submission is created
+        const mirageSubmission = server.create('collection-submission', {
+            id: guid.id,
+            creator: currentUser,
+            collection: mirageCollection,
+            guid,
+            reviewsState,
+        });
+
+        // And the collectionSubmission is stored in state
+        const submissionId = mirageSubmission.id + '-' + mirageSubmission.id;
+        const store = testContext.owner.lookup('service:store');
+        const collectionSubmission = await store.findRecord('collection-submission', submissionId);
+        testContext.set('collectionSubmission', collectionSubmission);
+    }
+
+    // Dynamic creating of all test cases based on the review states
+    for (const reviewsState of (Object.keys(collectionSubmissionTestCases) as CollectionSubmissionReviewStates[])) {
+        test(`shows actions for collection submission state: ${reviewsState}`,
+            async function(this: TestContext, assert) {
+                await buildCollectionScenarios(this, reviewsState);
```

Comment:
There's nothing wrong with how the test scenarios are prepopulated and this way works just as well, but for future reference, there is also a test module hook you can use like
```
hooks.beforeEach(() => {
    // test data setup code
});
```

If you want to set variables to use in your tests in the `beforeEach` hook, you'll have to define them in a new TestContext interface, similar to how [this test](https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/tests/acceptance/guid-file/registration-file-detail-test.ts#L17-L20) does it, or you can use `testContext.set('propertyName', propertyValue);` like you are doing in this test

Reply:
You can still use the `beforeEach` hook and reuse basically all of the code in `buildCollectionScenarios` but since it won't have access to the `reviewsState` you will have to just set `this.collectionSubmission.reviewsState = reviewsState` in the test itself, on line 109 or so. The way you have it right now works and is easy to read, so I don't think it warrants this small refactor though!